timely/src/dataflow/operators/generic/builder_raw.rs

```diff
@@ -161,6 +161,26 @@ impl<G: Scope> OperatorBuilder<G> {
         let inputs = self.shape.inputs;
         let outputs = self.shape.outputs;
 
+        // Generate a tracing span for the function.
+        #[cfg(feature = "tracing")]
+        let span = {
+            let location = std::panic::Location::caller();
+            // timely.console.span is a marker target allowing us to
+            // turn these spans on and off specifically
+            // TODO(guswynn): check with tracing folks if there is a better
+            // way to do this
+            tracing::trace_span!(
+                target: "timely.console.span",
+                "runtime.spawn",
+                kind = %"timely-operator",
+                "fn" = %std::any::type_name::<L>(),
+                task.name = self.shape.name.as_str(),
```

Comment:
i don't know anything about timely dataflow, so...is `shape_name` something provided by the user that can describe a particular task in the user code, or is it provided by the library? if it's internal to the library and describes a general class of dataflow operator, i would probably use that as the `kind` field rather than the `task.name` field --- in Tokio's instrumentation, the `kind` field will have values like "task", for tasks spawned via `tokio::spawn`, "local", for tasks spawned via `spawn_local`, "blocking" for blocking-pool tasks, and "block_on" for tasks created by `tokio::runtime::block_on`.

if `shape_name` is something internal to timely dataflow, i would probably use that for the `kind` field --- the target of the span will still indicate that it's coming from timely. if it's provided by the user, though, this seems correct.

Reply:
Yeah, it is something that people set if using the raw builder api, or wrappers around it like `unary_frontier`

For other combinators, like `flat_map`, `consolidate`, etc., its set to a uppercase version of the function name.
I think it would require more thorough changes throughout timely to distinguish between these, but using `kind` is a great point for a followup

things like
```
timely_stream.flat_map(|...| logic)
```
would end up with
```
kind = "timely-FlatMap"
name = "FlatMap"
```
(and possible named version of the same functions in the future
and things like
```
timely_stream.unary_frontier(p_contract, "my_special_name", || logic)
```
```
kind = "timely-UnaryFrontier"
name = "my_special_name"
```

I need to go through and add `#[track_caller]` to all these combinators anyways, so it might be nice to a do extra kind/name refactoring at the same time